qiskit/transpiler/passmanager.py

```diff
@@ -389,67 +389,68 @@ def __init__(self, stages=None, **kwargs):
             AttributeError: If a stage in the input keyword arguments is not defined.
             ValueError: If an invalid stage name is specified.
         """
-        if stages is None:
-            self.stages = [
-                "init",
-                "layout",
-                "routing",
-                "translation",
-                "optimization",
-                "scheduling",
-            ]
-        else:
-            invalid_stages = [
-                stage for stage in stages if self.invalid_stage_regex.search(stage) is not None
-            ]
-            if invalid_stages:
-                with io.StringIO() as msg:
-                    msg.write(f"The following stage names are not valid: {invalid_stages[0]}")
-                    for invalid_stage in invalid_stages[1:]:
-                        msg.write(f", {invalid_stage}")
-                    raise ValueError(msg.getvalue())
-
-            self.stages = stages
+        stages = stages or [
+            "init",
+            "layout",
+            "routing",
+            "translation",
+            "optimization",
+            "scheduling",
+        ]
+        self._validate_stages(stages)
+        super().__setattr__("_stages", tuple(stages))
+        super().__setattr__("_expanded_stages", tuple(self._generate_expanded_stages()))
```

Comment:
On the subject of pylint errors, would you mind suppressing them as `disable=no-attribute` (or whatever it is) by name - it's easier to tell when reading the file what's actually being suppressed.

Reply:
(Also, I'm pretty sure I fixed this false positive in pylint a little while ago - we just still see it because we use a pinned older version, as pylint has an unfortunate tendency to add questionable new errors with every release.)